File: Leetcode/typescript/1945-sum-string-digits.ts
function getLucky(s: string, k: number): number {
  let numStr = s
    .split("")
    .map((c) => (c.charCodeAt(0) - 96).toString())
    .join("");

  for (let i = 0; i < k; i++) {
    let sum = 0;
    for (const c of numStr) {
      sum += parseInt(c, 10);
    }
    numStr = sum.toString();
  }
  return parseInt(numStr, 10);
}

File: Leetcode/python/1945-sum-string-digits.py
class Solution:
    def getLucky(self, s: str, k: int) -> int:
        numStr = ''.join(str(ord(c) - 96) for c in s)

        res = 0
        for _ in range(k):
            res = sum(int(c) for c in numStr)
            numStr = str(res)
        return res
